Update OpenMote board header file to define UPDATE_CCA

aeneby opened the pull request:
This enables the cc2538's bootloader backdoor, to allow booting into the bootloader and flashing via the serial interface even when a valid image is already present. Backdoor enable is set to PORT A, PIN 6, (ON/SLEEP on the OpenBase), LOW for consistency with the default OpenMote firmware, patch based on existing code for same functionality in cc2538dk and ReMote boards.

aeneby commented:
Corresponding issue is #4432.

aeneby commented:
@katezilla Any feedback on the PR?

OlegHahm commented:
I re-assign to @haukepetersen, because I think he has the hardware in his office.

aeneby commented:
@haukepetersen do you have the kit available to be able to test this?

aeneby commented:
...anyone?

OlegHahm commented:
Sorry for the silence, @aeneby. @emmanuelsearch, you have the board now, right?

emmanuelsearch commented:
Yes I have the board, but no way to connect to it physically to my computer yet...

aeneby commented:
No problem @OlegHahm, I understand it's been busy. It would be good to know if someone does have the OpenMote board though, since I'll probably have more PRs to review in the near future if so.

kYc0o commented:
@emmanuelsearch so maybe you can test this now?

kYc0o commented:
I'm trying to test this but as the flashing script that we have is using the JLink flasher I'm not able to flash using the UART. How can I test this somehow?

aeneby commented:
Rebased to master and added parentheses as per e97af7386214b4fe61d47773bf5157333d10e7c8.

@kYc0o you have two options - either test with #4460 included, in which case ```make flash``` should flash via UART by default, or you can call the cc2538-bsl script manually with something like:

```python dist/tools/cc2538-bsl/cc2538-bsl.py -p "/dev/ttyUSB0" -e -w -v -b 115200 $HEXFILE```

Hope that helps.

kYc0o commented:
@aeneby I tried what you suggested and I successfully flashed something but the firmware... then the device was bricked... do you experienced something similar? any method to "unbrick" the device? I tried to erase all memory using JLinkExe on Mac OSX without success...

aeneby commented:
@kYc0o you should always be able to flash via JTAG. If you are using #4460 as well then just run ```make flash``` with ```PROGRAMMER=jlink```.

What do you mean by "bricked"? Please see https://github.com/RIOT-OS/RIOT/pull/4460#issuecomment-190878410 and see if that clarifies things.

kYc0o commented:
Ok I was trying to reflash the device without this PR merged so the device was blocked in a valid image status. I succeeded with flashing entering in the bootloader mode by grounding the ON/SLEEP pin, then resetting. It takes longtime to flash a simple hello-world though... I ACK however.

kYc0o commented:
Ok well... this is my first time with the new CI so I'm scared... anyone can just be sure that the test with the RIOT CI was successful?

authmillenon commented:
@kYc0o this week Travis has still preference.
